influxql/engine.go

```diff
@@ -669,54 +669,66 @@ func ReduceMean(key Key, values []interface{}, e *Emitter) {
 
 // MapMin collects the values to pass to the reducer
 func MapMin(itr Iterator, e *Emitter, tmin int64) {
-	var values []float64
+	var min *float64
 
 	for k, v := itr.Next(); k != 0; k, v = itr.Next() {
-		values = append(values, v.(float64))
+		if min == nil {
+			m := v.(float64)
+			min = &m
+		}
+		m := math.Min(*min, v.(float64))
+		min = &m
+	}
+	if min != nil {
+		e.Emit(Key{tmin, itr.Tags()}, *min)
 	}
-	e.Emit(Key{tmin, itr.Tags()}, values)
 }
 
 // ReduceMin computes the min of value.
 func ReduceMin(key Key, values []interface{}, e *Emitter) {
 	var min *float64
+
 	for _, value := range values {
-		vals := value.([]float64)
-		for _, v := range vals {
-			// Initialize min
-			if min == nil {
-				min = &v
-			}
-			m := math.Min(*min, v)
-			min = &m
+		v := value.(float64)
+		// Initialize min
+		if min == nil {
+			min = &v
 		}
+		m := math.Min(*min, v)
+		min = &m
 	}
 	e.Emit(key, min)
```

Comment:
Do you need to wrap this call to `Emit` in a `if` check for exactly the same reasons as the map phase? If every single node returned zero results, this too would have no results to emit, right?